core/src/test/java/org/frankframework/jdbc/StoredProcedureQuerySenderTest.java

```diff
@@ -32,46 +30,85 @@
 import org.frankframework.core.SenderResult;
 import org.frankframework.dbms.JdbcException;
 import org.frankframework.parameters.Parameter;
 import org.frankframework.stream.Message;
 import org.frankframework.testutil.TestFileUtils;
+import org.frankframework.testutil.TransactionManagerType;
+import org.frankframework.testutil.junit.DatabaseTest;
+import org.frankframework.testutil.junit.DatabaseTestEnvironment;
+import org.frankframework.testutil.junit.WithLiquibase;
+import org.frankframework.util.LogUtil;
 import org.frankframework.util.StreamUtil;
-import org.junit.After;
-import org.junit.Before;
-import org.junit.Test;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
 
-public class StoredProcedureQuerySenderTest extends JdbcTestBase {
+import liquibase.Contexts;
+import liquibase.Liquibase;
+import liquibase.database.Database;
+import liquibase.database.DatabaseFactory;
+import liquibase.database.jvm.JdbcConnection;
+import liquibase.resource.ClassLoaderResourceAccessor;
 
+@WithLiquibase(tableName = StoredProcedureQuerySenderTest.TABLE_NAME, file = "Jdbc/StoredProcedureQuerySender/DatabaseChangelog-StoredProcedures.xml")
+public class StoredProcedureQuerySenderTest {
+
+	protected static final String TABLE_NAME = "SP_TESTDATA";
 	public static final String TEST_DATA_STRING = "tést-data";
 	StoredProcedureQuerySender sender;
+	protected static Logger log = LogUtil.getLogger(StoredProcedureQuerySenderTest.class);
+	protected Liquibase liquibase;
 
 	PipeLineSession session;
 
-	@Before
-	public void setUp() throws Exception {
-		runMigrator("Jdbc/StoredProcedureQuerySender/DatabaseChangelog-StoredProcedures.xml");
+	@DatabaseTest.Parameter(0)
+	private TransactionManagerType transactionManagerType;
+
+	@DatabaseTest.Parameter(1)
+	private String dataSourceName;
+
+	@BeforeEach
+	public void setUp(DatabaseTestEnvironment databaseTestEnvironment) throws Throwable {
+		runMigrator("Jdbc/StoredProcedureQuerySender/DatabaseChangelog-StoredProcedures.xml", databaseTestEnvironment);
 
-		sender = getConfiguration().createBean(StoredProcedureQuerySender.class);
+		sender = databaseTestEnvironment.getConfiguration().createBean(StoredProcedureQuerySender.class);
 		sender.setSqlDialect("Oracle");
-		sender.setDatasourceName(productKey);
+		sender.setDatasourceName(dataSourceName);
 
 		session = new PipeLineSession();
 	}
 
-	@After
-	public void tearDown() {
+	@AfterEach
+	public void tearDown(DatabaseTestEnvironment databaseTestEnvironment) throws Throwable {
 		if (session != null) {
 			session.close();
 		}
+		databaseTestEnvironment.getConnection().close();
 	}
 
-	@Test
-	public void testSimpleStoredProcedureNoResultNoParameters() throws Exception {
-		assumeThat("H2 driver gives incorrect results for this test case", productKey, is(not("H2")));
+	protected void runMigrator(String changeLogFile, DatabaseTestEnvironment databaseTestEnvironment) throws Throwable {
```

Comment:
You shouldn't have to do this. The annotation `@WithLiquibase` should take care of all of this.